Sources/RediStack/Configuration.swift

```diff
@@ -234,8 +267,34 @@ extension RedisConnectionPool {
             connectionPassword: String? = nil,
             connectionDefaultLogger: Logger? = nil,
             tcpClient: ClientBootstrap? = nil
+        ) {
+            self.init(
+                connectionInitialDatabase: connectionInitialDatabase,
+                connectionUsername: nil,
+                connectionPassword: connectionPassword,
+                connectionDefaultLogger: connectionDefaultLogger,
+                tcpClient: tcpClient
+            )
+        }
+
+        /// Creates a new connection factory configuration with the provided options.
+        /// - Parameters:
+        ///     - connectionInitialDatabase: The optional database index to initially connect to. The default is `nil`.
+        ///       Redis by default opens connections against index `0`, so only set this value if the desired default is not `0`.
+        ///     - connectionUsername: The optional username to authenticate connections with. The default is `nil`. Works only with Redis 6 and greater.
+        ///     - connectionPassword: The optional password to authenticate connections with. The default is `nil`.
+        ///     - connectionDefaultLogger: The optional prototype logger to use as the default logger instance when generating logs from connections.
+        ///       If one is not provided, one will be generated. See `RedisLogging.baseConnectionLogger`.
+        ///     - tcpClient: If you have chosen to configure a `NIO.ClientBootstrap` yourself, this will be used instead of the `.makeRedisTCPClient` factory instance.
+        public init(
+            connectionInitialDatabase: Int? = nil,
+            connectionUsername: String? = nil,
+            connectionPassword: String? = nil,
+            connectionDefaultLogger: Logger? = nil,
+            tcpClient: ClientBootstrap? = nil
```

Comment:
I think this will break, as we now have two instances of `.init()` where all arguments are defaulted to `nil`